uniffi_bindgen/src/bindings/kotlin/templates/macros.kt

```diff
@@ -134,10 +134,15 @@ v{{- field_num -}}
 
 // Macro for destroying fields
 {%- macro destroy_fields(member) %}
-    Disposable.destroy(
     {%- for field in member.fields() %}
-        this.{{ field.name()|var_name }}{%- if !loop.last %}, {% endif -%}
-    {% endfor -%})
+        {%- if field.name().is_empty() %} 
+            // No Label
+        {% else %}
+            Disposable.destroy(
+            this.{{ field.name()|var_name }}{%- if !loop.last %}, {% endif -%}
```

Comment:
I don't think you can skip destroying the field in the above case.  Instead, you can use the `field_name` macro which works in both cases.  See [the enum template](https://github.com/mhammond/uniffi-rs/blob/a98523b664064ca80bb1a53a8425a191b01ed9ad/uniffi_bindgen/src/bindings/kotlin/templates/EnumTemplate.kt#L117) for an example.  Since you're already in the macro template, you don't need the `kt::` prefix.

Reply:
Thank you for your input @bendk. Since I haven't used Kotlin myself I wasn't fully aware and made a incorrect assumption.

Would it be correct to assume that if a Enum `Sealed Class` contain a variant with a field, all variants for that specific Enum (SC) should generate a `destroy` function? For simple Enums (without associated data) no `destroy` is needed to clear up pointer references because they're generated as `enum class` in Kotlin?

I updated my implementation but just want to understand if I'm following the correct assumptions:
```
// Macro for destroying fields
{%- macro destroy_fields(member) %}
    {%- for field in member.fields() %}
            Disposable.destroy(
        {%- if field.name().is_empty() %} 
            this.{%- call field_name(field, loop.index) -%}
        {% else %}
            this.{{ field.name()|var_name }}{%- if !loop.last %}, {% endif -%}
        {% endif %}
            )
    {% endfor -%}
{%- endmacro -%}
```

Reply:
That's correct.  I think you can make it even simpler though, since `field_name` works for both cases.:

```
 // Macro for destroying fields
{%- macro destroy_fields(member) %}
    {%- for field in member.fields() %}
            Disposable.destroy(this.{%- call field_name(field, loop.index) -%})
    {% endfor -%}
{%- endmacro -%}
```

Reply:
That worked like a charm, updated it in my latest push.